app/components/Calendar.tsx

```diff
@@ -1,58 +1,29 @@
 import moment from 'moment';
-import { useState, useCallback } from 'react';
+import { useState, useCallback, useEffect } from 'react';
 import { View } from 'react-big-calendar';
 import { CalendarEventProps, CalEvent } from '~/utils/interfaces';
 import EventModal from '~/components/common/eventModal';
 import RdsCalendar from '~/components/common/rdsCalendar';
-
-const initialEventsList: CalEvent[] = [
-  {
-    id: 1,
-    title: 'timed event',
-    start: moment().add(1, 'hour').toDate(),
-    end: moment().add(2, 'hours').toDate(),
-    description: '',
-    location: '',
-    visibility: 'private',
-    attendees: [{ attendee: { email: 'a@b.c' } }, { attendee: { email: 'alpha@beta.gamma' } }],
-  },
-  {
-    id: 2,
-    title: 'Some title',
-    start: moment().add(1, 'day').toDate(),
-    end: moment().add(1, 'day').add(2, 'hour').toDate(),
-    description: '',
-    location: '',
-    visibility: 'public',
-    attendees: [
-      { attendee: { email: 'alpha@beta.gamma' } },
-      { attendee: { email: 'admin@rds.com' } },
-    ],
-  },
-  {
-    id: 3,
-    title: 'Another title',
-    start: moment().add(1, 'day').add(8, 'hours').toDate(),
-    end: moment().add(1, 'day').add(9, 'hours').toDate(),
-    description: '',
-    location: '',
-    visibility: 'private',
-    attendees: [{ attendee: { email: 'admin@rds.com' } }],
-  },
-];
+import { useStore } from '~/store/useStore';
 
 interface CalendarProps {
   view?: View;
 }
 
 const Calendar = ({ view }: CalendarProps) => {
-  const [eventsList, setEventsList] = useState<CalEvent[]>(initialEventsList);
+  const events = useStore((state) => state.events);
+
+  const [eventsList, setEventsList] = useState<CalEvent[]>(events);
```

Comment:
why are we passing events, as initial state, because we are already setting state, on mount with events

Reply:
passing events in initial state will ready the page with existing events.
The useEffect part helps in two other cases:
1. When the store gets events for the first time,
2. When the events array is modified

Reply:
@isVivek99  Yes, that is because I'm setting the events in `useEffect` in `index.tsx` route.

I still don't see it as a problem because initial state does not trigger a re-render. So it's not actually an issue.